paimon-core/src/main/java/org/apache/paimon/partition/actions/HttpReportMarkDoneAction.java

```diff
@@ -0,0 +1,179 @@
+
+    public HttpReportMarkDoneAction(FileStoreTable fileStoreTable, CoreOptions options) {
+
+        Preconditions.checkArgument(
+                !StringUtils.isNullOrWhitespaceOnly(options.httpReportMarkDoneActionUrl()),
+                String.format(
+                        "Parameter %s must be non-empty when you use `http-report` partition mark done action.",
+                        PARTITION_MARK_DONE_ACTION_URL.key()));
+
+        this.fileStoreTable = fileStoreTable;
+        this.params = options.httpReportMarkDoneActionParams();
+
+        HttpClientOptions httpClientOptions =
+                new HttpClientOptions(
+                        options.httpReportMarkDoneActionUrl(),
+                        Optional.of(options.httpReportMarkDoneActionTimeout()),
+                        Optional.of(options.httpReportMarkDoneActionTimeout()),
+                        RESTObjectMapper.create(),
+                        1,
+                        DefaultErrorHandler.getInstance());
+        this.client = new HttpClient(httpClientOptions);
+    }
+
+    @Override
+    public void markDone(String partition) throws Exception {
+        HttpReportMarkDoneResponse response =
+                client.post(
+                        null,
+                        new HttpReportMarkDoneRequest(
+                                params,
+                                fileStoreTable.fullName(),
+                                fileStoreTable.location().toString(),
+                                partition),
+                        HttpReportMarkDoneResponse.class,
+                        Collections.emptyMap());
+        Preconditions.checkArgument(
+                reportIsSuccess(response),
+                String.format(
+                        "The http-report actions response attribute `result` should be 'SUCCESS' but is '%s'.",
+                        response.getResult()));
+    }
+
+    public boolean reportIsSuccess(HttpReportMarkDoneResponse response) {
+        return response != null && RESPONSE_SUCCESS.equalsIgnoreCase(response.getResult());
+    }
+
+    @Override
+    public void close() throws IOException {
+        try {
+            this.client.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /** RestRequest only for HttpReportMarkDoneAction. */
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class HttpReportMarkDoneRequest implements RESTRequest {
+        private static final String MARK_DONE_PARTITION = "partition";
+        private static final String TABLE = "table";
+        private static final String PATH = "path";
+        private static final String PARAMS = "params";
+
+        @JsonProperty(MARK_DONE_PARTITION)
+        private final String partition;
+
+        @JsonProperty(TABLE)
+        private final String table;
+
+        @JsonProperty(PATH)
+        private final String path;
```

Comment:
nullable?

Reply:
Whether it can be null is determined by the remote server.